org.mixedrealitytoolkit.uxcore/Experimental/NonNativeKeyboard/NonNativeKeyTouchAdapter.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+using System.Collections;
+using MixedReality.Toolkit;
+using MixedReality.Toolkit.Input;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.UX.Experimental
+{
+    public class NonNativeKeyTouchAdapter : MonoBehaviour
+    {
+        private const float ColliderMargin = 30.0f;
+        private const float ColliderThickness = 70.0f;
+        private const float ColliderZDelta = 20.0f;
+        private const float ReClickDelayTime = 1.0f;
+        private const float AnimationTime = 0.2f;
+        private const float AnimationMovementDelta = 20.0f;
+
+        private StatefulInteractable interactable;
+        private Graphic image;
+        private float lastClickTime;
+        private bool isInitialized;
+        private Vector3 defaultPosition;
+        private Vector3 animatedPosition;
+        private BoxCollider buttonCollider;
+        private Vector3 buttonColliderDefaultCenter;
+        private Color defaultImageColor;
+
+        private void Awake()
```

Comment:
As a best practive we are requiring that MonoBehaviour event functions be at least "protected", if the class is not sealed.

This requirement is meant to prevent subclasses from accidentally hiding the base's implementation of event functions (Awake, OnEnable, ect.).  That is, if the base event functions are private and the subclass implemented its own version, the compiler will happily allow this (no warning, no error).   This would then hide the base implementation. 

If the base event functions are protected, the compiler will at least provide a warning message.

So please make `Awake` , `OnEnable`, and `Update` projected.

Reply:
Done